R/DownloadModule.R

```diff
@@ -50,15 +50,16 @@ download_report_button_srv <- function(id,
                                        ),
                                        rmd_yaml_args = list(
                                          author = "NEST", title = "Report",
-                                         date = as.character(Sys.Date()), output = "html_document"
+                                         date = as.character(Sys.Date()), output = "html_document",
+                                         toc = FALSE
                                        )) {
   checkmate::assert_class(reporter, "Reporter")
   checkmate::assert_subset(rmd_output, c(
     "html_document", "pdf_document",
     "powerpoint_presentation", "word_document"
   ))
   checkmate::assert_list(rmd_yaml_args, names = "named")
-  checkmate::assert_true(all(c("author", "title", "date", "output") %in% names(rmd_yaml_args)))
+  checkmate::assert_true(all(c("author", "title", "date", "output", "toc") %in% names(rmd_yaml_args)))
```

Comment:
Does this check whether supplied names are legal or whether all mandatory features have been specified?

Reply:
now we demand all of them but in future we want to support a subset from them.

Reply:
Then I suggest changing 
```checkmate::assert_list(rmd_yaml_args, names = "named")``` 
to 
```checkmate::assert_list(rmd_yaml_args, names = "named", any.missing = FALSE)``` 
and adding type assertions on elements of `rmd_yaml_args`.

Reply:
good call but it is out of the scope of this issue. please create a new issue to check the type for each element of rmd_ymal_args.